validator/tests/mock.rs

```diff
@@ -176,6 +184,204 @@ async fn validators_can_do_skip_block() {
+
+    let validators = build_validators::<Network>(
+        env,
+        &(5u64..=10u64).collect::<Vec<_>>(),
+        &mut None,
+        false,
+        true,
+    )
+    .await;
+
+    // Listen for blockchain events from the new block producer (after a skip block).
+    let validator = validators.first().unwrap();
+    let validator_proxy = validator.proxy();
+
+    let validator = validators.last().unwrap();
+    let blockchain = Arc::clone(&validator.blockchain);
+    let events = blockchain.read().notifier_as_stream();
+
+    for validator in validators {
+        log::info!("Spawning validator: {}", validator.validator_address());
+        spawn(validator);
+    }
+
+    validator_proxy
+        .validator_health
+        .write()
+        ._set_publish_flag(false);
+
+    events.take(30).for_each(|_| future::ready(())).await;
+
+    let current_validator_health = validator_proxy.validator_health.read().health();
+
+    match current_validator_health {
+        ValidatorHealthState::Yellow => {
+            log::info!("Current validator health is yellow, as expected",)
+        }
+        _ => panic!("Validator Health different than expected"),
+    };
+
+    let events = blockchain.read().notifier_as_stream();
+
+    // Now we produce more blocks, and the validator should be inactivated again
+    events.take(20).for_each(|_| future::ready(())).await;
+
+    let current_validator_health = validator_proxy.validator_health.read().health();
+
+    match current_validator_health {
+        ValidatorHealthState::Red => {
+            log::info!("Current validator health is red, as expected",)
+        }
+        _ => panic!("Validator Health different than expected"),
+    };
+}
+
+#[test(tokio::test)]
+async fn validator_health_fully_recover() {
+    let env =
+        MdbxDatabase::new_volatile(Default::default()).expect("Could not open a volatile database");
+
+    let validators = build_validators::<Network>(
+        env,
+        &(5u64..=10u64).collect::<Vec<_>>(),
+        &mut None,
+        false,
+        true,
+    )
+    .await;
+
+    // Listen for blockchain events from the new block producer (after a skip block).
+    let validator = validators.first().unwrap();
+    let validator_proxy = validator.proxy();
+    let validator_address = validator.validator_address();
+
+    log::info!(
+        "Listening to blockchain events from validator {} ",
+        validator_address,
+    );
+
+    let validator = validators.last().unwrap();
+    let blockchain = Arc::clone(&validator.blockchain);
+
+    let events = blockchain.read().notifier_as_stream();
+
+    for validator in validators {
+        log::info!("Spawning validator: {}", validator.validator_address());
+        spawn(validator);
+    }
+
+    validator_proxy
+        .validator_health
+        .write()
+        ._set_publish_flag(false);
+
+    events.take(30).for_each(|_| future::ready(())).await;
```

Comment:
I guess you can take 50 here and get in the red state immediately. Testing if the validator got the yellow status is already covered in another test.